lua/lspconfig/psalm.lua

```diff
@@ -0,0 +1,28 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+local server_name = 'psalm'
+local bin_name = 'bin/psalm-language-server'
+
+configs[server_name] = {
+  default_config = {
+    cmd = { bin_name },
+    filetypes = { 'php' },
+    root_dir = function(pattern)
+      local cwd = vim.loop.cwd()
+      local root = util.root_pattern('psalm.xml', 'composer.json', '.git')(pattern)
+      -- prefer cwd if root is a descendant
```

Comment:
Why is this preferred? I feel like there's just too many edge cases that will resolve to an incorrect root.

For example if `cwd` is the root of a monorepo, but the Psalm project is a few directories deep. Or if I'm opening a file from while standing in my home directory (I'm quite sure I don't want the root to resolve to `~`).